### PR TITLE
Fixed #819 and #821

### LINKED ISSUE #819: Weapon magazines not being dropped after unloading
# Preliminary information

## Category
BUG

## Bug: Release or Development?
0.2.2a5

# Report
## Current Behavior
In title

## Expected/Wanted/Requested Behavior
They should be dropped at least if they are not empty

## Steps to Reproduce
Please enter the steps to reproduce the bug or behaviour:
1. Pick up a charged weapon. Make a shot to make sure it has ammo
2. Press R to unload a weapon
3. Try to make a shot to make sure it's unloaded. No ammo clips will be visible

## Network
Test server

### PR DESCRIPTION
**Pull request by JacobSweeten**
### Purpose
Fixes #819 and #821

### Approach
Hours of agony and over-complicating the problem in my mind.

Mag drop issue: I finally found the code where the magazine is added to the player's equipment pool (It was staring me in the face). It ran serverside, but it asked for the local player. I fixed this.

The incorrect ammo issue was fixed with a check.

#821 hand breaking bug was caused by client code calling server code. Made a workaround,

### Open Questions and Pre-Merge TODOs

- [x]  I read the contribution guidelines and the wiki.
- [X]  This fix is tested on the same branch it is PR'ed to.
- [X]  I correctly commented my code
- [X]  This PR does not include files without specific need to do so.
- [X]  This PR does not bring up any new compile errors
- [X]  This PR has been tested in editor
- [X]  This PR has been tested in multiplayer

### Notes:
None

### CODE REVIEW

**Comment by JacobSweeten** [comment]
There is one issue Doobly found. Should be a quick fix.

**Comment by JacobSweeten** [comment]
Dropping weapon drops clip too.

**Comment by DooblyNoobly** [comment]
give this a go and test it out:

`public void OnRemoveFromPool()
         {
            if (CurrentMagazine != null)
            {
                EquipmentPool.DisposeOfObject(ControlledByPlayer, CurrentMagazine.gameObject);
            }
             ControlledByPlayer = NetworkInstanceId.Invalid;
         }`

**Comment by JacobSweeten** [comment]
I couldn't access Equipment pool from weapon.cs and I wanted to add consistency so I made a new PNA overload of RemoveFromEquipmentPool()

File: UnityProject/Assets/Scripts/Weapons/Weapon.cs [hunk 1]
```diff
@@ -307,9 +308,8 @@ private void AttemptToFireWeapon()
 		private void Reload(GameObject m, string hand)
 		{
 			Debug.Log("Reloading");
-			PlayerManager.LocalPlayerScript.weaponNetworkActions.CmdLoadMagazine(gameObject, m);
+			PlayerManager.LocalPlayerScript.weaponNetworkActions.CmdLoadMagazine(gameObject, m, hand);
```
[the remark this answers is not in the document]
Comment (JacobSweeten) [reply UnityProject/Assets/Scripts/Weapons/Weapon.cs:312]: Should be client

**Comment by JacobSweeten** [comment]
We should be all ready to test and merge.

**Comment by DooblyNoobly** [comment]
just waiting on @BeastKiller2013 to clean up that PNA method then it should be good to merge

**Pull request merged**